MapboxCoreNavigation/RouteProgress.swift

```diff
@@ -157,8 +157,14 @@ open class RouteProgress: NSObject {
             
             if let segmentCongestionLevels = leg.segmentCongestionLevels, let expectedSegmentTravelTimes = leg.expectedSegmentTravelTimes  {
                 
-                for step in leg.steps.dropFirst() {
-                    let stepCoordinatesCount = maneuverCoordinateIndex + Int(step.coordinateCount) - 1
+                for step in leg.steps {
+                    guard let coordinates = step.coordinates else { continue }
+                    let stepCoordinates = step.maneuverType == .arrive ? Int(step.coordinateCount) : coordinates.dropLast().count
+                    let stepCoordinatesCount = maneuverCoordinateIndex + stepCoordinates - 1
```

Comment:
This isn’t the count of coordinates but rather an index into `segmentCongestionLevels` that corresponds to the last coordinate of this step and the first coordinate (and maneuver) of the following step. A better name would be `nextManeuverCoordinateIndex`.